mage/ginkgo/lib.go

```diff
@@ -103,11 +102,11 @@ func (k *NsWrapper) countStep() int {
 	return n
 }
 
-func (k NsWrapper) Terminate() error {
+func (ns NsWrapper) Terminate() {
```

Comment:
why stop returning the error?

Reply:
It looks like the return value is being ignored at least most of the time by the callers.  Still, maybe we should add panics at the caller instead of ignoring things.  I am not sure what is the best approach here.  What is the consequence of an error during Terminate() ?

Reply:
Yeah this is a tricky one. In general the functions here in the ginko util are in charge of doing a ginko friendly panic (usually `Expect`) as a way to have to put in less error handling logic in the tests themselves. 

As Ryan mentioned, we were always ignoring the return value of this, so it seemed weird to keep returning it. I'm not sure what the best way to handle the errors here is. We could query the cluster before each delete call here to make sure that a resource exists before attempting to delete, in case that causes an error that we don't care about (I think it does?). We also don't want to panic immediately, because we want to try and clean up as much as we can (probably?). 

We could perhaps keep a list of errors that occurred and at the end do something with them. Failing to cleanup is a bigger deal now with helm since subsequent runs will just fail out completely if any resources remain, and being in a partially cleaned up state will also prevent us from being able to do a `helm uninstall` on a subsequent run to resolve it.

Reply:
yeah I noticed we have a more graceful uninstall... maybe give this function a comment like it's the shutdown handler, and make sure the graceful one is error checking?

my concern is sometimes not fully cleaning up leaks resources that cost $$$, like PVs

Reply:
Helm will uninstall everything except for datacenters, crd, namespace itself. If any of helm's installed resources are not there anymore, it will fail to uninstall everything.

So helm is really only one chunk of the important cleanup here. It seems like we should probably track if any of the steps failed, cleanup as much as we can, and then report any errors?

```
1) Delete datacenters and store any errors
2) Helm uninstall, and store any errors
3) Delete crd
4) Delete namespace, and store any errors <--- though this step will probably still hang indefinitely if step 1 failed...maybe we shouldn't attempt it if step 1 failed.
```

Reply:
^ looks right to me - but if it's gonna be more than a few more lines of changes you can punt it to another PR